Deceiver.ts

```diff
@@ -22,27 +21,29 @@ export class DeceiverMirror<T, K extends keyof T> {
     }
 
     public getMethodNames (): K[] {
-        return _(this.getAllPrototypes())
+        return _.chain(this.getAllPrototypes())
             .map(prototype => ({prototype, names: Object.getOwnPropertyNames(prototype)}))
             .map(({prototype, names}) => ({
                 prototype,
                 names: names.filter(name => typeof Object.getOwnPropertyDescriptor(prototype, name).value == 'function'),
             }))
             .flatMap(({names}) => names)
             .filter(name => name != 'constructor')
-            .reduce(this.toUniqueArray, []);
+            .reduce(this.toUniqueArray, [])
+            .value();
     }
 
     public getPropertyNames (): K[] {
-        return _(this.getAllPrototypes())
+        return _.chain(this.getAllPrototypes())
             .map(prototype => ({prototype, names: Object.getOwnPropertyNames(prototype)}))
             .map(({prototype, names}) => ({
                 prototype,
                 names: names.filter(name => Object.getOwnPropertyDescriptor(prototype, name).get),
             }))
             .flatMap(({names}) => names)
             .filter(name => name !== '__proto__')
-            .reduce(this.toUniqueArray, []);
+            .reduce(this.toUniqueArray, [])
+            .value();
```

Comment:
These methods: `getMethodNames` and `getPropertyNames` - look very similar to each other, is there any way to abstract them?

I think the only unique lines in both are these:
```.filter(name => name !== '__proto__')``` vs ```.filter(name => name != 'constructor')```
```names: names.filter(name => typeof Object.getOwnPropertyDescriptor(prototype, name).value == 'function'),``` vs  ```names: names.filter(name => Object.getOwnPropertyDescriptor(prototype, name).get),
              }))		              }))```

Reply:
I will check how this code could look like on branch with memoization